ably.ts.d

```diff
@@ -0,0 +1,457 @@
+    export type SUSPENDED = "suspended";
+    export type FAILED = "failed";
+  }
+  type ChannelState = ChannelState.FAILED | ChannelState.INITIALIZED | ChannelState.SUSPENDED | ChannelState.ATTACHED | ChannelState.ATTACHING | ChannelState.DETACHED | ChannelState.DETACHING;
+
+  namespace ConnectionState {
+    export type INITIALIZED = "initialized";
+    export type CONNECTING = "connecting";
+    export type CONNECTED = "connected";
+    export type DISCONNECTED = "disconnected";
+    export type SUSPENDED = "suspended";
+    export type CLOSING = "closing";
+    export type CLOSED = "closed";
+    export type FAILED = "failed";
+  }
+  type ConnectionState = ConnectionState.INITIALIZED | ConnectionState.CONNECTED | ConnectionState.CONNECTING | ConnectionState.DISCONNECTED | ConnectionState.SUSPENDED | ConnectionState.CLOSED | ConnectionState.CLOSING | ConnectionState.FAILED;
+
+  namespace ConnectionEvent {
+    export type INITIALIZED = "initialized";
+    export type CONNECTING = "connecting";
+    export type CONNECTED = "connected";
+    export type DISCONNECTED = "disconnected";
+    export type SUSPENDED = "suspended";
+    export type CLOSING = "closing";
+    export type CLOSED = "closed";
+    export type FAILED = "failed";
+    export type UPDATE = "update";
+  }
+  type ConnectionEvent = ConnectionEvent.INITIALIZED | ConnectionEvent.CONNECTED | ConnectionEvent.CONNECTING | ConnectionEvent.DISCONNECTED | ConnectionEvent.SUSPENDED | ConnectionEvent.CLOSED | ConnectionEvent.CLOSING | ConnectionEvent.FAILED | ConnectionEvent.UPDATE;
+
+  namespace PresenceAction {
+    export type ABSENT = "absent";
+    export type PRESENT = "present";
+    export type ENTER = "enter";
+    export type LEAVE = "leave";
+    export type UPDATE = "update";
+  }
+  type PresenceAction = PresenceAction.ABSENT | PresenceAction.PRESENT | PresenceAction.ENTER | PresenceAction.LEAVE | PresenceAction.UPDATE;
+
+  namespace StatsIntervalGranularity {
+    export type MINUTE = "minute";
+    export type HOUR = "hour";
+    export type DAY = "day";
+    export type MONTH = "month";
+  }
+  type StatsIntervalGranularity = StatsIntervalGranularity.MINUTE | StatsIntervalGranularity.HOUR | StatsIntervalGranularity.DAY | StatsIntervalGranularity.MONTH;
+
+  namespace ProtocolMessageAction {
+    export type HEARTBEAT = "heartbeat";
+    export type ACK = "ack";
+    export type NACK = "nack";
+    export type CONNECT = "connect";
+    export type CONNECTED = "connected";
+    export type DISCONNECT = "disconnect";
+    export type DISCONNECTED = "disconnected";
+    export type CLOSE = "close";
+    export type CLOSED = "closed";
+    export type ERROR = "error";
+    export type ATTACH = "attach";
+    export type ATTACHED = "attached";
+    export type DETACH = "detach";
+    export type DETACHED = "detached";
+    export type PRESENCE = "presence";
+    export type MESSAGE = "message";
+    export type SYNC = "sync";
+    export type AUTH = "auth";
+  }
```

Comment:
This is all wrong (they're integers, not strings), but luckily not needed as not a public api